prov/terraform/google/compute_instance.go

```diff
@@ -0,0 +1,325 @@
+	consulKeys := commons.ConsulKeys{Keys: []commons.ConsulKey{}}
+
+	// Define the private IP address using the value exported by Terraform
+	privateIP := fmt.Sprintf("${google_compute_instance.%s.network_interface.0.address}",
+		instance.Name)
+
+	consulKeyPrivateAddr := commons.ConsulKey{
+		Path:  path.Join(instancesKey, instanceName, "/attributes/private_address"),
+		Value: privateIP}
+
+	consulKeys.Keys = append(consulKeys.Keys, consulKeyPrivateAddr)
+
+	// Define the public IP using the value exported by Terraform
+	// except if it was specified the instance shouldn't have a public address
+	var accessIP string
+	if instance.NoAddress {
+		accessIP = privateIP
+	} else {
+		accessIP = fmt.Sprintf("${google_compute_instance.%s.network_interface.0.access_config.0.assigned_nat_ip}",
+			instance.Name)
+		consulKeyPublicAddr := commons.ConsulKey{
+			Path:  path.Join(instancesKey, instanceName, "/attributes/public_address"),
+			Value: accessIP}
+		// For backward compatibility...
+		consulKeyPublicIPAddr := commons.ConsulKey{
+			Path:  path.Join(instancesKey, instanceName, "/attributes/public_ip_address"),
+			Value: accessIP}
+
+		consulKeys.Keys = append(consulKeys.Keys, consulKeyPublicAddr,
+			consulKeyPublicIPAddr)
+	}
+
+	// IP Address capability
+	capabilityIPAddr := commons.ConsulKey{
+		Path:  path.Join(instancesKey, instanceName, "/capabilities/endpoint/attributes/ip_address"),
+		Value: accessIP}
+	// Default TOSCA Attributes
+	consulKeyIPAddr := commons.ConsulKey{
+		Path:  path.Join(instancesKey, instanceName, "/attributes/ip_address"),
+		Value: accessIP}
+
+	consulKeys.Keys = append(consulKeys.Keys, consulKeyIPAddr, capabilityIPAddr)
+
+	commons.AddResource(infrastructure, "consul_keys", instance.Name, &consulKeys)
+
+	// Check the connection in order to be sure that ansible will be able to log on the instance
+	nullResource := commons.Resource{}
+	re := commons.RemoteExec{Inline: []string{`echo "connected"`},
+		Connection: &commons.Connection{User: user, Host: accessIP,
+			PrivateKey: `${file("` + privateKeyFilePath + `")}`}}
+	nullResource.Provisioners = make([]map[string]interface{}, 0)
+	provMap := make(map[string]interface{})
+	provMap["remote-exec"] = re
+	nullResource.Provisioners = append(nullResource.Provisioners, provMap)
+
+	commons.AddResource(infrastructure, "null_resource", instance.Name+"-ConnectionCheck", &nullResource)
+
+	return nil
+}
+
+func getStringProperty(kv *api.KV, deploymentID, nodeName, propertyName string, mandatory bool) (string, error) {
+
+	var result string
+	var err error
+	if _, result, err = deployments.GetNodeProperty(kv, deploymentID,
+		nodeName, propertyName); err != nil {
+		return result, err
+	}
+
+	if result == "" && mandatory {
+		return result, errors.Errorf("Missing value for mandatory parameter %s of %s",
+			propertyName, nodeName)
+	}
+
+	return result, nil
+}
+
+func getBooleanProperty(kv *api.KV, deploymentID, nodeName, propertyName string) (bool, error) {
```

Comment:
It could be great we can use these functions for other provisioners in a generic package as deployments ?

Reply:
ok will move these unctions